app/uk/gov/hmrc/tariffclassificationfrontend/controllers/AuthenticatedAction.scala

```diff
@@ -16,17 +16,50 @@
 
 package uk.gov.hmrc.tariffclassificationfrontend.controllers
 
+import javax.inject.{Inject, Singleton}
+import play.api.mvc.Results._
 import play.api.mvc.{ActionBuilder, Request, Result}
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.auth.core.AuthProvider.PrivilegedApplication
+import uk.gov.hmrc.auth.core._
+import uk.gov.hmrc.auth.core.retrieve.Retrievals
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.play.bootstrap.config.AuthRedirects
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.connector.StrideAuthConnector
 import uk.gov.hmrc.tariffclassificationfrontend.models.{AuthenticatedRequest, Operator}
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-object AuthenticatedAction extends ActionBuilder[AuthenticatedRequest] {
+@Singleton
+class AuthenticatedAction @Inject()(appConfig: AppConfig,
+                                    override val config: Configuration,
+                                    override val env: Environment,
+                                    override val authConnector: StrideAuthConnector)
+  extends ActionBuilder[AuthenticatedRequest]
+    with AuthorisedFunctions
+    with AuthRedirects {
+
+  lazy val enrolment = Enrolment(appConfig.authEnrolment)
 
   override def invokeBlock[A](request: Request[A], block: AuthenticatedRequest[A] => Future[Result]): Future[Result] = {
-    // TODO Implement Authentication Check based on the `request`. See DIT-311.
-    // This currently hard-codes the current operator with operatorId "0"
-    block(AuthenticatedRequest(Operator("0"), request))
-  }
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(
+      request.headers,
+      Some(request.session)
+    )
 
+    authorised(enrolment and AuthProviders(PrivilegedApplication))
+      .retrieve(Retrievals.credentials and Retrievals.name) {
+        retrieved =>
+          val id = retrieved.a.providerId
+          val name = retrieved.b.name
+          block(AuthenticatedRequest(Operator(id, name), request))
+      } recover {
+      case _: NoActiveSession => toStrideLogin(s"http://${request.host}${request.uri}")
+      case _: AuthorisationException => Redirect(routes.SecurityController.unauthorized())
+      case e => throw e
```

Comment:
```
case e => throw e
```
does not add anything.
If you remove it, you have the same behaviour because not handled by the `recover`

Reply:
Sorted